packages/documentsearch/test/searchmodel.spec.ts

```diff
@@ -61,6 +62,18 @@ describe('documentsearch/searchmodel', () => {
       });
     });
 
+    describe('#parsingError', () => {
+      it('should set informative string message on invalid regex', async () => {
+        model.useRegex = true;
+        expect(model.parsingError).toEqual('');
+        model.searchExpression = 'query\\';
+        await signalToPromise(model.stateChanged);
```

Comment:
It can happen that the signal is already emitted prior to the connection done in `signalToPromise`. So I would advice the safer:

```suggestion
        const waitForStateChanged = signalToPromise(model.stateChanged);
        model.searchExpression = 'query\\';
        await waitForStateChanged;
```